### PR TITLE
Nacos Cloud Service Registry add group name attribute

### PR DESCRIPTION
**Pull request by seas521**
I tested in a simple way ,please check it

### CODE REVIEW

File: spring-cloud-alibaba-nacos-discovery/src/main/java/org/springframework/cloud/alibaba/nacos/NacosDiscoveryProperties.java [hunk 1]
```diff
@@ -381,6 +396,11 @@ public void overrideFromEnv(Environment env) {
 			this.setEndpoint(
 					env.resolvePlaceholders("${spring.cloud.nacos.discovery.endpoint:}"));
 		}
+		if (StringUtils.isEmpty(this.getGroupName())) {
+			this.setEndpoint(
```
Comment (fangjian0423) [inline spring-cloud-alibaba-nacos-discovery/src/main/java/org/springframework/cloud/alibaba/nacos/NacosDiscoveryProperties.java]: It seems should invoke this.setGroupName(...)

Comment (seas521) [reply spring-cloud-alibaba-nacos-discovery/src/main/java/org/springframework/cloud/alibaba/nacos/NacosDiscoveryProperties.java]: My bad! pushed it again

**Comment by flystar32** [comment]
@seas521  thank you for contribution to Spring-Cloud-Alibaba

I've asked @nkorange(https://github.com/nkorange), he said when you register a service with group, then you can only subscribe the service in the same group.

how do you think about this?

if this is the right way, we may use 

```
 List<Instance> selectInstances(String serviceName, String groupName, boolean healthy) throws NacosException;
```
in `NacosServerList.getServers()`

```
	private List<NacosServer> getServers() {
		try {
			List<Instance> instances = discoveryProperties.namingServiceInstance()
					.selectInstances(serviceId, true);
			return instancesToServerList(instances);
		}
		catch (Exception e) {
			throw new IllegalStateException(
					"Can not get service instances from nacos, serviceId=" + serviceId,
					e);
		}
	}
```

**Comment by seas521** [comment]
> String

I think the group name is designed for setting apart from instances in same-name services，dont know if I am getting wrong,I just check the latest version of nacos-client1.0

**Comment by fangjian0423** [comment]
I think group is a little bit repetitive like namespace.

If you register services with specified group, you should subscribe services with the same group.  

Now we could not subscribe all services in same namespace with all groups. If nacos support it, maybe this pull request can be merged.

**Comment by chuntaojun** [comment]
> I think group is a little bit repetitive like namespace.
> 
> If you register services with specified group, you should subscribe services with the same group.
> 
> Now we could not subscribe all services in same namespace with all groups. If nacos support it, maybe this pull request can be merged.

if let user desc serviceName like this "serviceName@@group" and nacos can auto analyze group info，I think it can work

**Pull request closed without merging**